plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/PredicatePushdownController.java

```diff
@@ -16,9 +16,15 @@
 import io.trino.spi.connector.ConnectorSession;
 import io.trino.spi.predicate.DiscreteValues;
 import io.trino.spi.predicate.Domain;
+import io.trino.spi.predicate.Marker;
```

Comment:
I did not read the code yet but statement in commit message for `Fix CASE_INSENSITIVE_PUSHDOWN` is false (or I do not understand what you are trying to say).

> Both are equal, while 'A' is lower and 'a' is higher ('A' < 'a').

In MySQL both are equal - true, but the other part of the sentence is not true. Neither is larger than the other:

```
select 'A'<'a';
+---------+
| 'A'<'a' |
+---------+
|       0 |
+---------+

select 'a'>'A';
+---------+
| 'a'>'A' |
+---------+
|       0 |
+---------+
```